File: 038.count_and_say/count_and_say.cpp
class Solution {
public:
    string ReadOff(string &s) {
        string tmp = "";
        char prev = '0';
        int counter = 0;
        for (int i = 0; i < s.length(); i++) {
            if (prev == '0' || prev == s[i]) {
                prev = s[i];
                counter++;
            } else {
                tmp += to_string(counter);
                tmp += prev;
                prev = s[i];
                counter = 1;
            }
        }
        tmp += to_string(counter);
        tmp += prev;
        return tmp;
    }

    string countAndSay(int n) {
        string ans = "1";
        if (n == 1) return ans;
        while (n > 1) {
            ans = ReadOff(ans);
            n--;
        }
        return ans;
    }
};

File: 038.count_and_say/count_and_say.py
class Solution(object):
    def readOff(self, s):
        ret, pre, t = "", None, 0
        for i in xrange(len(s)):
            if (s[i] != pre):
                if (pre != None):
                    ret += str(t) + str(pre)
                t, pre = 1, s[i]
            else:
                t += 1
        ret += str(t) + str(pre)
        return ret

    def countAndSay(self, n):
        s = "1"
        while n > 1:
            n, s = n - 1, self.readOff(s)
        return s

